### PR TITLE
Fix/ui tests

### PR DESCRIPTION
**Pull request by AdityaHarvi**
#### Summary of changes/features added:
- Tests are cooler now.
- Added the requests-for-book tests which also includes searching and notifications all in one.

#### Screenshots of UI changes (update with each new commit): 
- n/a

#### Link to referenced github issue: 
m/a

## Pull Request Checklist
- [ ] Unit test written?
- [x] (**For later in project**) Integration tests written?
- [x] Java docs written?

### CODE REVIEW

**Review by rjmah1 — APPROVED** [review]
lgtm

File: client/app/src/androidTest/java/com/example/criengine/TestUtilityMethods.java [hunk 1]
```diff
@@ -2,16 +2,66 @@
...
+        // Returns True if you can find "My Books" on the screen. Waits 50 seconds to find
+        // at least 1 match. This is to counter potentially long wait times when logging in.
+        assertTrue(solo.waitForText("My Books", 1, 50000));
+    }
+
+    /**
+     * Log in with the second intentTesting account.
+     * @param solo The solo object.
+     */
+    public static void login2(Solo solo) {
```
Comment (ducklin5) [inline client/app/src/androidTest/java/com/example/criengine/TestUtilityMethods.java:41]: login2 is the same as login1, can we pass the email and password as an argument?

**Review by ducklin5 — APPROVED** [review]
Lgtm. Only comment was the login functions but thats up to you.

File: client/app/src/androidTest/java/com/example/criengine/TestUtilityMethods.java (hunk 1 above)
Comment (AdityaHarvi) [reply client/app/src/androidTest/java/com/example/criengine/TestUtilityMethods.java]: fixed

**Pull request merged**